src/Compatibility/Microsoft.DotNet.ApiCompatibility/Abstractions/CompatDifference.cs

```diff
@@ -0,0 +1,88 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.CodeAnalysis;
+using System;
+
+namespace Microsoft.DotNet.ApiCompatibility.Abstractions
+{
+    /// <summary>
+    /// Class representing a difference of compatibility, containing detailed information about it.
+    /// </summary>
+    public class CompatDifference : IDiagnostic, IEquatable<CompatDifference>
+    {
+        /// <summary>
+        /// The Diagnostic ID for this difference.
+        /// </summary>
+        public string DiagnosticId { get; }
+
+        /// <summary>
+        /// The <see cref="DifferenceType"/>.
+        /// </summary>
+        public DifferenceType Type { get; }
+
+        /// <summary>
+        /// A diagnostic message for the difference.
+        /// </summary>
+        public virtual string Message { get; }
+
+        /// <summary>
+        /// A unique ID in order to identify the API that the difference was raised for.
+        /// </summary>
+        public string ReferenceId { get; }
+
+        private CompatDifference() { }
+
+        /// <summary>
+        /// Instanciates a new object representing the compatibility difference.
+        /// </summary>
+        /// <param name="id"><see cref="string"/> representing the diagnostic ID.</param>
+        /// <param name="message"><see cref="string"/> message describing the difference.</param>
+        /// <param name="type"><see cref="DifferenceType"/> to describe the type of the difference.</param>
+        /// <param name="member"><see cref="ISymbol"/> for which the difference is associated to.</param>
+        public CompatDifference(string id, string message, DifferenceType type, ISymbol member)
+            : this(id, message, type, member.GetDocumentationCommentId())
```

Comment:
do we need to add a null check here for the ```member``` ?

Reply:
Good catch.